core/Pos/Core/Vss/Types.hs

```diff
@@ -60,4 +62,12 @@ instance Hashable VssCertificate where
 
 -- | VssCertificatesMap contains all valid certificates collected
 -- during some period of time.
-type VssCertificatesMap = HashMap StakeholderId VssCertificate
+--
+-- Invariants:
+--   * stakeholder ids correspond to 'vcSigningKey's of associated certs
+--   * no two certs have the same 'vcVssKey'
+newtype VssCertificatesMap = UnsafeVssCertificatesMap
+    { getVssCertificatesMap :: HashMap StakeholderId VssCertificate }
+    deriving (Eq, Show, Generic, NFData, Monoid, Container, NontrivialContainer)
```

Comment:
`Monoid` looks wrong to me, `mappend` is dangerous, because it can break one invariant.

Reply:
I'm going to fix `Monoid` instead of removing it entirely, because we use `<>` in some places.